test/AllUserSleep-tests.js

```diff
@@ -1,26 +1,171 @@
+        "sleepQuality": 2.5
+      },
+      {
+        "userID": 2,
+        "date": "2019/06/30",
+        "hoursSlept": 9.5,
+        "sleepQuality": 3.9
+      },
+      {
+        "userID": 3,
+        "date": "2019/06/30",
+        "hoursSlept": 6.7,
+        "sleepQuality": 1.1
+      },
+      {
+        "userID": 1,
+        "date": "2019/07/01",
+        "hoursSlept": 2.8,
+        "sleepQuality": 4.3
+      },
+      {
+        "userID": 2,
+        "date": "2019/07/01",
+        "hoursSlept": 9,
+        "sleepQuality": 4.7
+      },
+      {
+        "userID": 3,
+        "date": "2019/07/01",
+        "hoursSlept": 8.1,
+        "sleepQuality": 1.8
+      },
+      {
+        "userID": 1,
+        "date": "2019/07/02",
+        "hoursSlept": 5.7,
+        "sleepQuality": 1.7
+      },
+      {
+        "userID": 2,
+        "date": "2019/07/02",
+        "hoursSlept": 9,
+        "sleepQuality": 3.4
+      },
+      {
+        "userID": 3,
+        "date": "2019/07/02",
+        "hoursSlept": 4.3,
+        "sleepQuality": 3.9
+      },
+      {
+        "userID": 1,
+        "date": "2019/07/03",
+        "hoursSlept": 10.1,
+        "sleepQuality": 2.1
+      },
+      {
+        "userID": 2,
+        "date": "2019/07/03",
+        "hoursSlept": 4.2,
+        "sleepQuality": 4.7
+      },
+      {
+        "userID": 3,
+        "date": "2019/07/03",
+        "hoursSlept": 9.2,
+        "sleepQuality": 1.3
+      },
+      {
+        "userID": 1,
+        "date": "2019/07/04",
+        "hoursSlept": 6.9,
+        "sleepQuality": 2.5
+      },
+      {
+        "userID": 2,
+        "date": "2019/07/04",
+        "hoursSlept": 10.6,
+        "sleepQuality": 3
+      },
+      {
+        "userID": 3,
+        "date": "2019/07/04",
+        "hoursSlept": 5.3,
+        "sleepQuality": 4.1
+      },
+      {
+        "userID": 1,
+        "date": "2019/07/05",
+        "hoursSlept": 8.8,
+        "sleepQuality": 1.7
+      },
+      {
+        "userID": 2,
+        "date": "2019/07/05",
+        "hoursSlept": 7.9,
+        "sleepQuality": 4.2
+      },
+      {
+        "userID": 3,
+        "date": "2019/07/05",
+        "hoursSlept": 6.1,
+        "sleepQuality": 2.3
+      },
+      {
+        "userID": 1,
+        "date": "2019/07/06",
+        "hoursSlept": 6.5,
+        "sleepQuality": 1.4
+      },
+      {
+        "userID": 2,
+        "date": "2019/07/06",
+        "hoursSlept": 7.7,
+        "sleepQuality": 4.5
+      },
+      {
+        "userID": 3,
+        "date": "2019/07/06",
+        "hoursSlept": 8.1,
+        "sleepQuality": 2.8
+      }];
+    allUserSleep = new AllUserSleep(sleepData);
+  });
+  it("Should have a parameter to take in user data", () => {
+    expect(allUserSleep.sleepData).to.equal(sleepData);
+  });
+  it("Should calculate the average sleep quality of all users", () => {
+    const avgSleep = allUserSleep.calcAvgSleepQuality(sleepData);
+    expect(avgSleep).to.equal(3.0625000000000004)
```

Comment:
I would add a little bit of code into your implementation here to round this number to 2 decimal points! User's won't really learn anything useful about this statistic beyond the hundredths place & it'll help standardize units of measurement across your display!

`.toFixed()` comes to mind!